Prox/Prox/Models/AppState.swift

```diff
@@ -6,19 +6,37 @@ import Foundation
 
 class AppState {
     enum State: String {
-        case loading, details, carousel, background, exiting, unknown
+        case initial, loading, permissions, details, carousel, background, unknown
     }
 
     private static var cardVisits = Set<Int>()
 
     // Initial app state
-    private static var state = State.loading
+    private static var state = State.initial
     private static var preBackgroundState: State?
 
     static func getState() -> State {
         return state
     }
 
+    static func enterLoading() {
+        updateSessionState(newState: State.loading)
+    }
+
+    static func requestPermissions() {
+        if (state == State.loading) {
```

Comment:
This was originally so that showing the permissions dialog didn't count as "loading" time, but I ended up replacing this somewhat complicated logic with events that fire when we show a dialog reprompting the user (if they've denied location permissions). This might throw off the loading time a bit for the first session (when the user has to decide which permissions to grant Prox), but I think that's okay.